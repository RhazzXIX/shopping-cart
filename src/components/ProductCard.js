import { Link } from "react-router-dom";
import "../styles/ProductCard.css";
import { useState } from "react";

const ProductCard = (props) => {
  const { product, addItem } = props;
  const [quantity, setQuantity] = useState(0);

  const decrement = (e) => {
    setQuantity((num) => {
      if (Number(num) === 0) return 0;
      return Number(num) - 1;
    });
  };
  const increment = (e) => {
    setQuantity((num) => Number(num) + 1);
  };

  const handleChangeQuantity = (e) => {
    if (e.target.value === 0) return;
    setQuantity(e.target.value);
  };

  const addItems = () => {
    if (quantity === 0) return;
    const productToCart = {
      ...product,
      qty: quantity,
    };

    addItem(productToCart);
  };
  return (
    <div className="card">
      <img src={product.images[0]} alt={product.title} />
      <h4>{product.title}</h4>
      <p>{product.description}</p>
      <p>$ {product.price}.00</p>
      <div className="qtyContainer">
        <label htmlFor="qty">
          Qty:
          <input
            id="qty"
            type="text"
            inputMode="numeric"
            placeholder="0"
            onChange={handleChangeQuantity}
            value={quantity !== 0 ? quantity : ""}
          />
        </label>
        <button className="increment" onClick={increment}>
          +
        </button>
        <button className="decrement" onClick={decrement}>
          -
        </button>
      </div>
      <p className="totalCost">$ {product.price * quantity}.00</p>
      <button className="addToCart" onClick={addItems}>
        Add to Cart
      </button>
    </div>
  );
};

export default ProductCard;
