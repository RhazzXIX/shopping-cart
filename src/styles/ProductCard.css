div.card {
  border: 1px solid black;
  padding: 12px;
  border-radius: 8px;
  display: grid;
  grid-template-rows: 2fr 1fr auto auto;
  align-items: center;
  justify-items: center;
  width: 260px;
  height: 360px;
  gap: 8px;
}

div.card:hover {
  box-shadow: 0 0 8px black;
}

div.card > img {
  justify-self: center;
  align-self: center;
  border-radius: 8px;
  width: auto;
  height: auto;
  max-width: 200px;
  max-height: 200px;
}

div.card > h4 {
  font-size: 1rem;
}

div.card > p:first-of-type {
  font-size: 1.2rem;
}

div.card > button {
  width: 200px;
  align-self: center;
  justify-self: center;
}
