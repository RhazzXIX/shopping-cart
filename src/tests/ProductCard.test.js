import React from "react";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import ProductCard from "../components/ProductCard";

const productDetails = {
  title: "Black Motorbike",
  id: "214",
  price: 569,
  images: ["https://i.dummyjson.com/data/products/91/1.jpg"],
  description:
    "Engine Type:Wet sump, Single Cylinder, Four Stroke, Two Valves, Air Cooled with SOHC (Single Over Head Cam) Chain Drive Bore & Stroke:47.0 x 49.5 MM",
};

describe("Product cards", () => {
  it("Renders the card of the product", () => {
    const { container } = render(<ProductCard product={productDetails} />);
    expect(container).toMatchSnapshot();
  });
});
