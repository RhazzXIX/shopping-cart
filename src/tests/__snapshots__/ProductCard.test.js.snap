// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component Renders the card of the product 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Black Motorbike"
      src="https://i.dummyjson.com/data/products/91/1.jpg"
    />
    <h4>
      Black Motorbike
    </h4>
    <p>
      Engine Type:Wet sump, Single Cylinder, Four Stroke, Two Valves, Air Cooled with SOHC (Single Over Head Cam) Chain Drive Bore & Stroke:47.0 x 49.5 MM
    </p>
    <p>
      $ 
      569
      .00
    </p>
    <div
      class="qtyContainer"
    >
      <label
        for="qty"
      >
        Qty:
        <input
          id="qty"
          inputmode="numeric"
          placeholder="0"
          type="text"
          value=""
        />
      </label>
      <button
        class="increment"
      >
        +
      </button>
      <button
        class="decrement"
      >
        -
      </button>
    </div>
    <p
      class="totalCost"
    >
      $ 
      0
      .00
    </p>
    <button
      class="addToCart"
    >
      Add to Cart
    </button>
  </div>
</div>
`;
